tcl/sql.tcl

```diff
@@ -62,11 +62,27 @@ proc qc::sql_sort { args } {
     }
     set order_by_list {}
     for {set i 0} {$i<[llength $list]} {incr i} {
+
 	set this_item [lindex $list $i]
         set this_item [string trim $this_item]
-        set this_item [qc::db_quote_identifier $this_item]
+        set parts [split $this_item "."]
+        switch [llength $parts] {
+            1 {
+                set this_item [qc::db_quote_identifier $this_item]                
+            }
+            2 {
+                set table [qc::db_quote_identifier [lindex $parts 0]]
+                set column [qc::db_quote_identifier [lindex $parts 1]]
+                set this_item "${table}.${column}"
+            }
+            default {
+                error "Unable to parse identifier \"$this_item\""
+            }
+        }
+
 	set next_item [lindex $list [expr {$i+1}]]
         set next_item [string trim $next_item]
+
```

Comment:
OK, if you have checked this won't break existing usage then this it is
probably reasonable for the developer to work around this constraint

On 4 August 2016 at 16:09, Peter Chaplin <notifications@github.com> wrote:

> In tcl/sql.tcl
> <https://github.com/qcode-software/qcode-tcl/pull/409#discussion_r73540715>
> :
>
> >  	set next_item [lindex $list [expr {$i+1}]]
> >          set next_item [string trim $next_item]
> > +
>
> Yes, that would be a problem. I don't think we have any existing cases of
> that usage, though, and going forwards, I think escaping the existing
> quotes would be the correct/expected behaviour.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/qcode-software/qcode-tcl/pull/409/files/2033187dd55e598a72e791a80cf4529ea2057dce#r73540715>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AC8lGR_jbTNBAeN5ElsfnRCO-K7QlHO6ks5qcgCmgaJpZM4Jcyw3>
> .
>



-- 
Kind Regards,
Daniel Clark
Qcode Software Limited
www.qcode.co.uk
Tel: +44(0) 1463 896 482